debian/clearwater-auto-config-docker.init.d

```diff
@@ -0,0 +1,103 @@
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+### BEGIN INIT INFO
+# Provides:          clearwater-auto-config-docker
+# Required-Start:    $network $local_fs
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: clearwater-auto-config-docker
+# Description:       clearwater-auto-config-docker
+# X-Start-Before:    clearwater-infrastructure bono sprout homer homestead ellis restund
+### END INIT INFO
+
+# Changes in this command should be replicated in clearwater-auto-config-*.init.d
+do_auto_config()
+{
+  config=/etc/clearwater/config
+  if [ -f /etc/clearwater/force_ipv6 ]
+  then
+    # The sed expression finds the first IPv6
+    ip=$(hostname -I | sed -e 's/\(^\|^[0-9. ]* \)\([0-9A-Fa-f:]*\)\( .*$\|$\)/\2/g')
+  else
+    # The sed expression finds the first IPv4 address in the space-separate list of IPv4 and IPv6 addresses.
+    # If there are no IPv4 addresses it finds the first IPv6 address.
+    ip=$(hostname -I | sed -e 's/\(^\|^[0-9A-Fa-f: ]* \)\([0-9.][0-9.]*\)\( .*$\|$\)/\2/g' -e 's/\(^\)\(^[0-9A-Fa-f:]*\)\( .*$\|$\)/\2/g')
+  fi
+
+  # Add square brackets around the address iff it is an IPv6 address
+  bracketed_ip=$(python /usr/share/clearwater/bin/bracket_ipv6_address.py $ip)
+
+  [ "$SPROUT_NAME" != "" ]    && sprout_hostname=$SPROUT_PORT_5054_TCP_ADDR                  || sprout_hostname=$ip
```

Comment:
Where might the `*_NAME` variables come from?  Are we expecting people to run

    sudo SPROUT_NAME=sprout.override.com service clearwater-auto-config-docker restart

or is this interfacing with some form of docker meta-data?